Added basic login box to popup that authenticates browser with extension backend

Bubba74 opened the pull request:
This is a pretty barebones development. It is mostly to demonstrate one possible user interface for the backend authentication mechanism.

See this pull request in the *-Backend repo for an example of how to authenticate: [link](https://github.com/BruinAssistant/Bruin-Assistant-Backend/pull/1#issuecomment-962678818)

I would greatly appreciate any feedback of how we plan on authenticating the user and stuff like that.

Bubba74 committed:
Added basic login box to popup that authenticates browser with extension backend

Bubba74 committed:
Added basic login box to popup that authenticates browser with extension backend

Clumsyndicate commented:
Do you think you could change this to use the Heroku app instances instead of your personal server?

Clumsyndicate committed:
Automatically check auth

Clumsyndicate committed:
Automatically check auth

Bubba74 committed:
Send/accept friend requests and view friends

Bubba74 committed:
Send/accept friend requests and view friends

Bubba74 committed:
Merge branch 'user-auth' of https://github.com/BruinAssistant/Bruin-Assistant-Extension into user-auth

Clumsyndicate committed:
Fix docs directory

Clumsyndicate committed:
Observer changes

New mutation observer only observes attribute change of the body.

Clumsyndicate committed:
Email and password checker

Bubba74 committed:
Merge branch 'user-auth' of https://github.com/BruinAssistant/Bruin-Assistant-Extension into user-auth

Bubba74 committed:
Refactored topbar browser window into many different files

The extension does many things to the point where some .js files
are highly related while others essentially do not interact. Even
those that do interact are largely independent of each other.

Those html/css/js files used in the topbar popup have been moved to
the /topbar directory. Further directories for html/css/js have also
been added.

Scripts that are used by both the topbar and other components of
the extension have been moved to /common. For example, config.js is
loaded both by groupme.js and /topbar/html/login.js.

Bubba74 committed:
Merge branch 'master' into user-auth

Bubba74 committed:
Remove more references to molasses
